src/lio/warmer_helpers.h

```diff
@@ -0,0 +1,220 @@
+/*
+   Copyright 2016 Vanderbilt University
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+//***********************************************************************
+// Warmer helper definitions
+//***********************************************************************
+
+
+#ifndef _WARMER_H_
+#define _WARMER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define WFE_SUCCESS   1
+#define WFE_FAIL      2
+#define WFE_WRITE_ERR 4
+
+typedef struct {
+    int  state;
+    int  nfailed;
+    char name[1];
+} __attribute__((packed)) db_inode_entry_t;
+
+typedef struct {
+    ex_id_t inode;
+    int  state;
+} __attribute__((packed)) db_rid_entry_t;
+
+//*************************************************************************
+// warm_put_inode - Puts an entry in the inode DB
+//*************************************************************************
+
+static int warm_put_inode(leveldb_t *db, ex_id_t inode, int state, int nfailed, char *name)
+{
+    leveldb_writeoptions_t *wopt;
+    db_inode_entry_t *r;
+    char *errstr = NULL;
+    int n;
+
+    n = strlen(name) + sizeof(db_inode_entry_t);
+    r = malloc(n);
+    r->state = state;
+    r->nfailed = nfailed;
+    strcpy(r->name, name);
+
+    wopt = leveldb_writeoptions_create();
+    leveldb_put(db, wopt, (const char *)&inode, sizeof(ex_id_t), (const char *)r, n, &errstr);
+    leveldb_writeoptions_destroy(wopt);
+
+    if (errstr != NULL) {
+        log_printf(0, "ERROR: %s\n", errstr);
+        free(errstr);
+    }
+
+    free(r);
+
+    return((errstr == NULL) ? 0 : 1);
+}
+
+//*************************************************************************
+// warm_put_rid - Puts an entry in the RID DB
+//*************************************************************************
+
+static int warm_put_rid(leveldb_t *db, char *rid, ex_id_t inode, int state)
```

Comment:
You can put the `__attribute__((unused))` here in the declaration to keep from having to put it in the consumers' code.